### ISSUE TITLE
Unpack without metadata.

### ISSUE THREAD

**Issue by kevincox** [alexcrichton/tar-rs#165]
I'm writing some software which unpacks not-quite-trusted archives and I would like to do this with with a very minimal interface. In my use case I only unpack files and only want to preserve names and data. I won't want to copy (or attempt to copy) things such as timestamps and xattrs. It would be nice if there was a way to do this using tar-rs.

Right now I can use `unpack_in` which provides "safe" unpacking but copies metadata by default. There are opt-out functions such as `set_preserve_permissions` and `set_unpack_xattrs` but blacklisting things to copy is a dangerous way to approach this problem. Furthermore I don't see a way to avoid copying the mtime.

I can manually unpack a file using the `Read` implementation however now I need to redo all of the safety checks that `unpack_in` does.

The TL;DR is that it would be nice if there was a way to either:
- Have a function to make all metadata copying default to off.
- Add a function which allows users to implement something like `unpack_in` without rewriting the safety logic.

**Comment by NightRa** [comment]
I also think that the metadata writing should be separated nicely in the library - as it's causing a huge increase in the number of handles opened compared to just extracting (writing) the files.
Something akin to `Entry::write_file` and `Entry::write_metadata`, with `Entry::unpack` retaining the current behavior.

* Note that this impacts performance 5x times, but only when there's no antivirus scanning the files, which dominates the runtime completely, in which case the metadata write time is negligible.

**Comment by alexcrichton** [comment]
I'd be down for a method like this! I think it'd be great to have a method which performs all the validation checks this does and allows custom logic to follow that if necessary
